src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -905,6 +905,17 @@ void QgsWmsCapabilities::parseLayer( QDomElement const & e, QgsWmsLayerProperty&
 
         parseStyle( e1, styleProperty );
 
+        for ( int i = 0; i < layerProperty.style.size(); ++i )
+        {
+          if ( layerProperty.style[i].name == styleProperty.name )
```

Comment:
Also note in this case that layerProperty is a non-const reference, so the compiler will always call the non-const [] operator, which at @m-kuhn mentioned forces a detach. If layerProperty was const then the const [] operator will be called (no detach), but that doesn't apply here.